sqlglot/optimizer/eliminate_join_marks.py

```diff
@@ -0,0 +1,184 @@
+import typing as t
```

Comment:
i think eliminate_join_marks should be a transform in oracle, not in optimizer

Reply:
It makes more sense there, however it does mean you now have a reference to the optimize module from dialects (due to the use of scope). 

Where should the function live? below the class or as part of the oracle generator?

Reply:
i think ideally at oracle parse time to get rid of the old syntax

Reply:
I've moved the function to the oracle dialect module, but not integrated it into the generator or parser yet due to other comments raised.